tests/cbtest/gtest/test_uzfs.cc

```diff
@@ -773,12 +790,207 @@ TEST(uZFS, TestIOConnAcceptor) {
 	close(fd);
 }
 
-void execute_rebuild_test_case(const char *s, int test_case, zvol_rebuild_status_t status)
+void
+uzfs_mock_zvol_rebuild_dw_replica(void *arg)
+{
+	rebuild_thread_arg_t *rebuild_args = (rebuild_thread_arg_t *)arg;
+
+	struct sockaddr_in replica_ip;
+
+	int		rc = 0;
+	int		sfd = -1;
+	uint64_t	offset = 0;
+	uint64_t	checkpointed_ionum;
+	zvol_info_t	*zinfo = NULL;
+	zvol_state_t	*zvol_state;
+	zvol_io_cmd_t	*zio_cmd = NULL;
+	zvol_io_hdr_t 	hdr;
+	struct linger lo = { 1, 0 };
+	char zvol_name[MAXNAMELEN];
+
+	strncpy(rebuild_args->zvol_name, "vol3", MAXNAMELEN);
+	sfd = rebuild_args->fd;
+	zinfo = rebuild_args->zinfo;
+
+	if ((rc = setsockopt(sfd, SOL_SOCKET, SO_LINGER, &lo, sizeof (lo)))
+	    != 0) {
+		LOG_ERRNO("setsockopt failed");
+		goto exit;
+	}
+
+	bzero(&replica_ip, sizeof (replica_ip));
+	replica_ip.sin_family = AF_INET;
+	replica_ip.sin_addr.s_addr = inet_addr(rebuild_args->ip);
+	replica_ip.sin_port = htons(rebuild_args->port);
+
+	if ((rc = connect(sfd, (struct sockaddr *)&replica_ip,
+	    sizeof (replica_ip))) != 0) {
+		LOG_ERRNO("connect failed");
+		perror("connect");
+		goto exit;
+	}
+
+	if (rebuild_test_case == 1) {
+		rc =-1;
+		goto exit;
+	}
+
+send_hdr_again:
+	/* Set state in-progess state now */
+	checkpointed_ionum = uzfs_zvol_get_last_committed_io_no(zinfo->zv);
+	zvol_state = zinfo->zv;
+	bzero(&hdr, sizeof (hdr));
+	hdr.status = ZVOL_OP_STATUS_OK;
+	hdr.version = REPLICA_VERSION;
+	hdr.opcode = ZVOL_OPCODE_HANDSHAKE;
+	hdr.len = strlen(rebuild_args->zvol_name) + 1;
+	if (rebuild_test_case == 2)
+		hdr.opcode = ZVOL_OPCODE_WRITE;
+
+	rc = uzfs_zvol_socket_write(sfd, (char *)&hdr, sizeof (hdr));
+	if (rc != 0) {
+		LOG_ERR("Socket hdr write failed");
+		goto exit;
+	}
+
+	if (rebuild_test_case == 3) {
+		rc = -1;
+		goto exit;
+	}
+
+	strncpy(zvol_name, rebuild_args->zvol_name, hdr.len);
+	if (rebuild_test_case == 4)
+		strncpy(zvol_name, "X", hdr.len);
+
+	printf("Volname:%s\n", zvol_name);
```

Comment:
I would remove all printfs from the test suite. Ideally the output of the test suite should look like: test name - status and nothing more. Currently our test output is cluttered by other messages printed by zrepl itself, which is tolerable, but we should not add more messages to it.

Reply:
Removed it.